src/ApolloHooksMutation.re

```diff
@@ -36,72 +30,98 @@ type controlledVariantResult('a) =
+
+type mutationResult('a) = {. "data": 'a};
+
+[@bs.deriving abstract]
+type options('a) = {
+  [@bs.optional]
+  variables: Js.Json.t,
+  [@bs.optional]
+  mutation: option(ReasonApolloTypes.queryString),
+  [@bs.optional]
+  client: ApolloClient.generatedApolloClient,
+  [@bs.optional]
+  refetchQueries,
+  [@bs.optional]
+  awaitRefetchQueries: bool,
+  [@bs.optional]
+  update: (ApolloClient.generatedApolloClient, mutationResult('a)) => unit,
+};
 
-  type jsResult = {
-    .
-    "data": Js.Nullable.t(Js.Json.t),
-    "loading": bool,
-    "called": bool,
-    "error": Js.Nullable.t(error),
-  };
+type jsResult = {
+  .
+  "data": Js.Nullable.t(Js.Json.t),
+  "loading": bool,
+  "called": bool,
+  "error": Js.Nullable.t(error),
+};
 
-  type jsMutate = (. options) => Js.Promise.t(jsResult);
-  type mutation =
+type jsMutate('a) = (. options('a)) => Js.Promise.t(jsResult);
+type mutation('a, 'b) =
+  (
+    ~mutation: 'b,
+    ~client: ApolloClient.generatedApolloClient=?,
+    ~refetchQueries: refetchQueries=?,
+    ~awaitRefetchQueries: bool=?,
+    unit
+  ) =>
+  Js.Promise.t(controlledVariantResult('a));
+
+[@bs.module "@apollo/react-hooks"]
+external useMutationJs:
+  (. ReasonApolloTypes.queryString, options('a)) => (jsMutate('a), jsResult) =
+  "useMutation";
+
+exception Error(string);
+
+let useMutation:
+  type t.
     (
-      ~variables: Js.Json.t=?,
       ~client: ApolloClient.generatedApolloClient=?,
+      ~variables: Js.Json.t=?,
       ~refetchQueries: refetchQueries=?,
       ~awaitRefetchQueries: bool=?,
-      unit
+      ~update: (ApolloClient.generatedApolloClient, mutationResult('a)) =>
+               unit
+                 =?,
+      (module ApolloHooksTypes.Config with type t = t)
     ) =>
-    Js.Promise.t(controlledVariantResult(Config.t));
-
-  [@bs.module "@apollo/react-hooks"]
-  external useMutation:
-    (. ReasonApolloTypes.queryString, options) => (jsMutate, jsResult) =
-    "useMutation";
-
-  let use =
+    (
       (
-        ~variables=?,
-        ~client=?,
-        ~refetchQueries=?,
-        ~awaitRefetchQueries=?,
-        ~update=?,
-        (),
-      ) => {
+        ~variables: Js.Json.t=?,
+        ~client: ApolloClient.generatedApolloClient=?,
+        ~refetchQueries: refetchQueries=?,
+        ~awaitRefetchQueries: bool=?,
+        unit
+      ) =>
+      Js.Promise.t(controlledVariantResult(t)),
+      controlledVariantResult(t),
+      controlledResult(t),
+    ) =
+  (
+    ~client=?,
+    ~variables=?,
+    ~refetchQueries=?,
+    ~awaitRefetchQueries=?,
+    ~update=?,
+    (module Config),
+  ) => {
     let (jsMutate, jsResult) =
-      useMutation(.
+      useMutationJs(.
         gql(. Config.query),
         options(
-          ~variables?,
           ~client?,
+          ~variables?,
           ~refetchQueries?,
           ~awaitRefetchQueries?,
           ~update?,
           (),
         ),
       );
 
-    let mutate: mutation =
+    let mutate =
```

Comment:
Can we include the type for this one? I guess it will help catch errors if someone refactors the implementation and it will no longer comply with the actual `mutation` type?